File: src/java-sample/com/xtaudio/xt/sample/PrintSimple.java
package com.xtaudio.xt.sample;

import com.xtaudio.xt.*;
import com.xtaudio.xt.NativeTypes.*;

public class PrintSimple {

    public static void main() throws Exception {
        try(AutoCloseable audio = XtAudio.init(null, null, null)) {
            for(XtSystem system: XtAudio.getSystems()) {
                XtService service = XtAudio.getService(system);
                for(int d = 0; d < service.getDeviceCount(); d++)
                    try(XtDevice device = service.openDevice(d)) {
                        System.out.println(system + ": " + device);
                    }
            }
        }
    }
}

File: src/cli-sample/PrintSimple.cs
using System;

namespace Xt
{
	public static class PrintSimple
	{
		public static void Main()
		{
			using XtAudio audio = new XtAudio(null, IntPtr.Zero,  null);
			foreach(XtSystem system in XtAudio.GetSystems())
			{
				XtService service = XtAudio.GetService(system);
				for (int d = 0; d < service.GetDeviceCount(); d++)
					using (XtDevice device = service.OpenDevice(d))
						Console.WriteLine(system + ": " + device);
			}
		}
	}
}